fabric-screen-api-v1/src/main/java/net/fabricmc/fabric/api/client/screen/v1/FabricScreen.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.screen.v1;
+
+import java.util.List;
+
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.AbstractButtonWidget;
+import net.minecraft.client.render.item.ItemRenderer;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Provides access to additional context a screen can hold.
+ */
+@Environment(EnvType.CLIENT)
+public interface FabricScreen {
```

Comment:
I wonder if there is a better name for this?

Reply:
Originally this was `ScreenContext` a few commits ago.

`ScreenInfo` could work.

Reply:
This is really fabric screens as this is injected wholesale into screen class.

Reply:
Decided on `ScreenExtensions` in the end.